### DISCUSSION TITLE
output to `new-only` in dev and `full` in CI

### DISCUSSION THREAD

**Discussion by BrianHicks** [vercel/turborepo#7199] [Help] [answered]
I'd like to change our `outputMode` for all jobs to `new-only` in dev. I think it's a better default because it gives the developers I'm serving a better idea of the cache boosts they're getting, and gives me a better idea about where I'm getting cache misses or hits that I wasn't expecting.

Problem is, I'd still like to have the full logs in CI to be able to debug problems easily. Is there some environment variable, maybe, that I can set to change the log level for all `turbo` invocations?

**Comment by anthonyshew** [comment, marked as answer]
> Is there some environment variable?

There isn't - but this sounds like it would be quite useful. If you want to write an RFC (or convert this Discussion to one), I think there's a good chance it would get a good discussion.

The way that I would probably do this today would be [installing global `turbo`](https://turbo.build/repo/docs/installing#install-globally) into your CI machine and then writing your desired turbo command there. So, for example:
```
// package.json
{
  "scripts": {
    "quality": "turbo lint --output-logs=new-only"
  }
}
```

```
// Your CI file, something like .github/workflows/quality.yml
...
 run: turbo lint
 ...
```

**Reply by BrianHicks** [reply]
I'll write an RFC for this. Thanks!

**Reply by BrianHicks** [reply]
https://github.com/vercel/turbo/discussions/7324
